### PR TITLE
BBD-97: Add customUrlParams

### PR DESCRIPTION
**Pull request by Vic-Dev**

### CODE REVIEW

File: test/fixtures.ts [hunk 1]
```diff
@@ -119,9 +111,9 @@ export let COMBINED_REFINEMENTS = [
 ];
 
 export let CUSTOM_PARAMS_FROM_STRING = [
```
Comment (Vic-Dev) [inline test/fixtures.ts]: Not used anywhere, should we still have tests in query for custom url params now that its moved to bridge or take this out?

**Review by Vic-Dev — COMMENTED** [review]
hi vic-dev

File: src/capacitor/index.ts [hunk 2]
```diff
@@ -28,6 +28,7 @@ export interface FluxConfiguration extends QueryConfiguration {
     headers?: any;
     https?: boolean;
     timeout?: number;
+    customUrlParams?: any;
```
Comment (effervescentia) [inline src/capacitor/index.ts]: `queryParams`

File: src/core/bridge.ts [hunk 3]
```diff
@@ -68,11 +68,12 @@ export abstract class AbstractBridge {
   }
 
   private extractRequest(query: any): { request: Request; queryParams: any; } {
+    const queryParams = Object.assign({}, this.config['customUrlParams'], query.queryParams);
```
Comment (effervescentia) [inline src/core/bridge.ts]: shouldn't need string literal accessor? (`['customUrlParams']`)

Comment (effervescentia) [reply src/core/bridge.ts]: also, we can only get `query.queryParams` in the case where type is `object` and `query instanceof Query`

File: src/core/bridge.ts [hunk 4]
```diff
@@ -68,11 +68,12 @@ export abstract class AbstractBridge {
...
     switch (typeof query) {
-      case 'string': return { request: new Query(<string>query).build(), queryParams: {} };
+      case 'string': return { request: new Query(<string>query).build(), queryParams };
       case 'object': return query instanceof Query
-        ? { request: query.build(), queryParams: query.queryParams }
-        : { request: query, queryParams: {} };
+        ? { request: query.build(), queryParams }
```
Comment (effervescentia) [inline src/core/bridge.ts]: break into if statement to handle pulling `queryParams` from `query`

File: test/fixtures.ts (hunk 1 above)
Comment (effervescentia) [reply test/fixtures.ts]: @Vic-Dev you can remove this

**Review by effervescentia — CHANGES_REQUESTED** [review]
💰 🏭

File: src/core/bridge.ts [hunk 5]
```diff
@@ -68,11 +68,12 @@ export abstract class AbstractBridge {
...
 
   private extractRequest(query: any): { request: Request; queryParams: any; } {
+    const queryParams = Object.assign({}, this.config.queryParams);
     switch (typeof query) {
-      case 'string': return { request: new Query(<string>query).build(), queryParams: {} };
+      case 'string': return { request: new Query(<string>query).build(), queryParams };
       case 'object': return query instanceof Query
-        ? { request: query.build(), queryParams: query.queryParams }
-        : { request: query, queryParams: {} };
+        ? { request: query.build(), queryParams: Object.assign({}, queryParams, query.queryParams) }
```
Comment (effervescentia) [inline src/core/bridge.ts]: can just be `Object.assign(queryParams, query.queryParams)` as the original `queryParams` from line 71 already builds off of a new empty object

**Review by effervescentia — CHANGES_REQUESTED** [review]
otherwise 🔥

**Pull request merged**